src/ReachSets/ContinuousPost/BFFPS19/reach_helper.jl

```diff
@@ -0,0 +1,52 @@
+#getting variables for specified blocks
+function get_vars(blocks, partition)
+    result = Vector{Int}()
+    b_i = 1
+    for p_i in 1:length(partition)
+        if p_i == blocks[b_i]
+            result = vcat(result, collect(partition[p_i]))
+            b_i += 1
+            if b_i > length(blocks)
+                return result
+            end
+        end
+     end
+    return result
+end
+
+#getting projection of constraints list
+function get_constraints_list_projections(constraints, vars)
+    result = Vector{LazySet}()
+    for c in constraints
+        push!(result, LazySets.Approximations.project(c, vars))
+    end
+
+    return result
+end
+
+#check whether intersection with between cartesian product array and constraints list is empty
+function is_intersection_disjoint(ca::CartesianProductArray, constraints_list::Vector{LazySet})
+    for c in constraints_list
+        if !isdisjoint(ca, c)
+            return false
+        end
+    end
+    return true
+end
+
+#compose high-dimensional cartesian product array from 2 low-dimensional CPA's
+function combine_array(Xb::CartesianProductArray{N}, Xdiff::CartesianProductArray{N},
+        blocks::Vector{Int}, diff_blocks::Vector{Int}) where {N}
+    result_l = length(Xb.array) + length(Xdiff.array)
+    result = CartesianProductArray(result_l, N)
+
+     for i in 1:result_l
+        if in(i,blocks)
+            push!(result.array, Xb.array[findall(x->x == i, blocks)[1]])
+        else
+            push!(result.array, Xdiff.array[findall(x->x == i, diff_blocks)[1]])
```

Comment:
This can also be simplified (I also renamed the function):
```julia
function combine_cpas(cpa1::CartesianProductArray{N, S}, cpa2::CartesianProductArray{N, S},
                       blocks1::Vector{Int}, blocks2::Vector{Int}) where {N, S<:LazySet{N}}
    result = Vector{S}(undef, length(array(Xb)) + length(array(Xdiff)))
    result[blocks] = array(Xb)
    result[diff_blocks] = array(Xdiff)
    return CartesianProductArray(result)
end
```
If you know that the set types of the two CPAs are the same, you can include their type parameters and